Cython/Compiler/Nodes.py

```diff
@@ -9474,6 +9474,19 @@ def analyse_declarations(self, env):
                 error(self.pos, "Invalid keyword argument: %s" % kw)
             else:
                 setattr(self, kw, val)
+        if ('with_gil' in self.kwargs and 'nogil' in self.kwargs and
+                self.kwargs['with_gil'] and self.kwargs['nogil']):
+            # But nogil=False, with_gil=False is fine - it just means we neither
+            # explicitly release or acquire the GIL.
+            error(self.pos, "Contradictory settings for 'with_gil' and 'nogil'")
+        if self.with_gil:
+            # For now warn against this pretty heavily. I don't want to deal with too
+            # many bug reports at this stage.
+            warning(
+                self.pos,
+                "'with_gil' is experimental. Cython does almost nothing to ensure "
+                "thread safe use of Python variables (including reference counting) "
+                "and so it is completely your responsibility. Use with caution!", 999)
```

Comment:
I don't intend to keep this warning forever. But I think we need more control of PyObject variables in parallel blocks before we can really start recommending it.